create_kubernetes_sysext.sh

```diff
@@ -0,0 +1,99 @@
+  echo "A temporary directory named SYSEXTNAME in the current folder will be created and deleted again."
+  echo "All files in the sysext image will be owned by root."
+  echo "To use arm64 pass 'ARCH=arm64' as environment variable (current value is '${ARCH}')."
+  "${SCRIPTFOLDER}"/bake.sh --help
+  exit 1
+fi
+
+VERSION="$1"
+SYSEXTNAME="$2"
+
+rm -f kubectl kubeadm kubelet
+
+# install kubernetes binaries.
+# TODO: use arch variable
+curl -o kubectl -fsSL "https://dl.k8s.io/v${VERSION}/bin/linux/${ARCH}/kubectl"
+curl -o kubeadm -fsSL "https://dl.k8s.io/v${VERSION}/bin/linux/${ARCH}/kubeadm"
+curl -o kubelet -fsSL "https://dl.k8s.io/v${VERSION}/bin/linux/${ARCH}/kubelet"
+
+rm -rf "${SYSEXTNAME}"
+mkdir -p "${SYSEXTNAME}"/usr/bin
+mv kubectl kubeadm kubelet "${SYSEXTNAME}"/usr/bin
+
+chmod +x "${SYSEXTNAME}"/usr/bin/{kubectl,kubeadm,kubelet}
+
+# setup kubelet service.
+mkdir -p "${SYSEXTNAME}/usr/lib/systemd/system"
+cat > "${SYSEXTNAME}/usr/lib/systemd/system/kubelet.service" <<-'EOF'
+[Unit]
+Description=kubelet: The Kubernetes Node Agent
+Documentation=https://kubernetes.io/docs/home/
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+ExecStart=/usr/bin/kubelet
+Restart=always
+StartLimitInterval=0
+RestartSec=10
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+mkdir -p "${SYSEXTNAME}/usr/lib/systemd/system/kubelet.service.d"
+cat > "${SYSEXTNAME}/usr/lib/systemd/system/kubelet.service.d/10-kubeadm.conf" <<-'EOF'
+# Note: This dropin only works with kubeadm and kubelet v1.11+
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
+Environment="KUBELET_CONFIG_ARGS=--config=/var/lib/kubelet/config.yaml"
+# This is a file that "kubeadm init" and "kubeadm join" generates at runtime, populating the KUBELET_KUBEADM_ARGS variable dynamically
+EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
+# This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
+# the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
+EnvironmentFile=-/etc/sysconfig/kubelet
+ExecStart=
+ExecStartPre=/usr/bin/mkdir -p /opt/libexec /opt/libexec.work
+ExecStartPre=/usr/bin/cp -r /usr/local/bin/cni/ /opt/bin/cni
+ExecStartPre=/usr/bin/cp /usr/local/share/kubernetes-version /etc/kubernetes-version
+ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_CONFIG_ARGS $KUBELET_KUBEADM_ARGS $KUBELET_EXTRA_ARGS
+EOF
+
+mkdir -p "${SYSEXTNAME}/usr/local/share/"
+echo "v${VERSION}" > "${SYSEXTNAME}/usr/local/share/kubernetes-version"
+
+cat > "${SYSEXTNAME}/usr/lib/systemd/system/usr-libexec.mount" <<-'EOF'
+[Unit]
+Description=Kubernetes flex volume plugin directory
+
+[Mount]
+What=overlay
+Where=/usr/libexec
+Type=overlay
+Options=lowerdir=/usr/libexec,workdir=/opt/libexec.work,upperdir=/opt/libexec
```

Comment:
Do you plan to ship anything under `/usr/libexec`? Currently it's not the case, or?

Reply:
Ah, understood. I'm a bit concerned about using an overlay because the mounts under `/usr` are manged by systemd-sysext now. If we know what folders should exist and be writable we could ship symlink from `/usr/libexec/kubernetes/` to `/var/kubernetes/` (`/opt` is also not a good idea I think).

Reply:
By the way, there is this to join configurations: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/control-plane-flags/#patches

Reply:
With symlink I mean that `/usr/libexec/kubernetes/` would be a symlink to `/var/kubernetes/` and we ensure that `/var/kubernetes/` exists with a helper service. Then the folder is writable.